### ISSUE TITLE
Getting DeprecationWarning

### ISSUE THREAD

**Issue by tsabunkar** [blakehaswell/mongoose-unique-validator#83]
Getting this **warning**;  _collection.count is deprecated, and will be removed in a future version. Use collection.countDocuments or collection.estimatedDocumentCount instead_
I am just performing unique constraint validation on email property for my userSchema i.e-

```
const UserSchema = new Schema({
    firstName: {
        type: String,
        required: true
    },
    lastName: {
        type: String,
        required: true
    },
    password: {
        type: String,
        required: true
    },
    email: {
        type: String,
        required: true,
        unique: true 
    },
    messages: { 
        type: [{
            type: Schema.Types.ObjectId,
            ref:  'mess_collec' //ref->reference to another model
        }]
    }
});

UserSchema.plugin(uniqueValidator);

UserModel = mongoose.model('user_collection', UserSchema)

```

**Comment by viveleroi** [comment]
Which version of mongoose are you using?

**Comment by tsabunkar** [comment]
I am using the following version :
```
 "mongoose": "^5.2.1",
 "mongoose-unique-validator": "^2.0.1"
```

**Comment by ftrevo** [comment]
@viveleroi possible cause is the code below, originated by the deprecation of the method `count` on mongoose. See this [link](http://mongoosejs.com/docs/api.html#model_Model.count).

`model.where({ $and: conditions }).count((err, count) => {`

Solution would be changing `count` to `countDocuments`, but this could have some compatibility issues with mongoose versions (need to check the version that `countDocuments` was included.)

Want a pull request to fix the problem, considering the possible compatibility issue?

**Comment by MadRabbit** [comment]
same issue

**Comment by idanyadgar** [comment]
Same here.
(http://mongodb.github.io/node-mongodb-native/3.1/api/Collection.html#count)

**Comment by rledford** [comment]
This repo uses `countDocuments` but the npm package was published with the deprecated `count`. I've updated the source in my node_modules and the issue is resolved but the package will need to be republished with a version bump.

**Comment by viveleroi** [comment]
I was concerned about releasing an update with failing tests but we now know the tests are failing because of a test teardown issue. I've released this update, v2.0.2
